edu-services/gradebook-service/api/src/java/org/sakaiproject/service/gradebook/shared/GradeMappingDefinition.java

```diff
@@ -102,9 +103,10 @@ public String toString() {
 
 		// we only ever order by bottom percents now
 		final DoubleComparator doubleComparator = new DoubleComparator(gradeMap);
-		final Map<String, Double> rval = new TreeMap<>(doubleComparator);
-		rval.putAll(gradeMap);
-
+		final Map<String, Double> tMap = new TreeMap<>(doubleComparator);
+		tMap.putAll(gradeMap);
+		final Map<String, Double> rval = new LinkedHashMap<>(tMap);
```

Comment:
Can you just double check where this method is used as I commented in the javadoc for it (now out of date) that it was specifically a treemap and to ensure that was considered. There might be additional unnecessary casts down the track.

Reply:
Hi Steve

I've searched for a "cast" on all methods where the "sortGradeMapping" method is called and I've found no "cast". The "sortGradeMapping" method return is only used as a Map and no implementation is used. The "sortGradeMapping" is called in these methods:

- org.sakaiproject.component.gradebook.GradebookServiceHibernateImpl.getCourseGradeForStudents(String,
List<String>, Map<String, Double>)

- org.sakaiproject.gradebookng.tool.component.GbCourseGradeChart.getData(Map<String, Double>)

- org.sakaiproject.component.gradebook.GradebookServiceHibernateImpl.getGradebookGradeMappings (Set<GradeMapping>)

- org.sakaiproject.component.gradebook.GradebookServiceHibernateImpl.getGradebookInformation(String)

- org.sakaiproject.gradebookng.tool.panels.SettingsGradingSchemaPanel.getGradingSchemaEntries()

- org.sakaiproject.gradebookng.tool.panels.SettingsGradingSchemaPanel.refreshCourseGradeChart
(AjaxRequestTarget)